linked_list.py

```diff
@@ -0,0 +1,348 @@
+
+# Defines a node in the singly linked list
+class Node(object):
+    def __init__(self, value=None, next_node=None):
+        self.__data = None  # protected
+        self.next_node = next_node
+        self.__set_data_once(value)
+
+    def get_data(self):
+        if self.__data:
+            return self.__data
+        else:
+            return "Undefined"
+
+    def __set_data_once(self, value):
+        if self.__data == None:
+            self.__data = value
+        else:
+            return ("You may not modify data.  Please add a node.")
+
+# Defines the singly linked list
+class LinkedList(object):
+    def __init__(self):
+        self.__head = None # keep the head private. Not accessible outside this class
+
+    def __retrieve_head(self):

```

Comment:
Since you want to keep the head private, there shouldn't be a method to retrieve head.